user/super/com/google/gwt/emul/java/util/stream/Stream.java

```diff
@@ -246,6 +251,10 @@ public boolean tryAdvance(Consumer<? super T> action) {
     return StreamSupport.stream(spliterator, false);
   }
 
+  default List<T> toList() {
+    return Collections.unmodifiableList(this.collect(Collectors.toList()));
```

Comment:
The official documentation states that "The implementation in this interface returns a List produced as if by the following"

```java
Collections.unmodifiableList(new ArrayList<>(Arrays.asList(this.toArray())))
```
I'm not saying that your version is wrong. I just like to know if it is better or more efficient.
https://docs.oracle.com/en/java/javase/16/docs/api/java.base/java/util/stream/Stream.html#toList()

Reply:
I also was going to comment on either this or using our existing Collectors.toUnmodifiableList() (note that I'm wrong: that will not accept nulls, but Stream.toList() is permitted to hold nulls... but has the other same semantics about immutability).

To your question though, StreamImpl.toArray already calls collect(toList()):
https://github.com/gwtproject/gwt/blob/a5e649dec66f65f4e1cd0ecd7bffd3e706243750/user/super/com/google/gwt/emul/java/util/stream/StreamImpl.java#L567-L570

Effectively, we would be then building a list, to build an array, to wrap in a list, to copy to another list. This implementation is better, simply creating and appending to a list, then wrapping to make it unmodifiable.